examples/nested/values.yaml

```diff
@@ -0,0 +1,24 @@
+spire-server:
+  enabled: true
+  upstreamAuthority:
+    spire:
+      enabled: true
+      server:
+        address: spire-server.spire-root-server
+  controllerManager:
+    enabled: true
+    identities:
+      spiffeIDTemplate: spiffe://{{ .TrustDomain }}/k8s/{{ .ClusterName }}/ns/{{ .PodMeta.Namespace }}/sa/{{ .PodSpec.ServiceAccountName }}
+
+spiffe-oidc-discovery-provider:
+  enabled: true
+  insecureScheme:
+    enabled: true
+
+upstream-spiffe-csi-driver:
+  enabled: true
```

Comment:
This may need some thought to properly handle enablement / disablement based on the upstream spire CA being enabled.  We don't want to advertise that it can be enabled in the values.yaml when the upstream spire CA is disabled, which indicates that a control flag designed this way is wrong, as it allows mismatches of settings that lead to nonsensical deployments.

I would do a Boolean truth table to be constructed, indicating the valid settings and then the values.yaml to be modified to have settings that only permit the valid combinations to be reached.

Reply:
Has to be done this way due to limitations in how helm chart nesting works. Helm isn't flexible enough to do that unfortunately.

Reply:
Done